libsolidity/formal/SMTEncoder.cpp

```diff
@@ -748,6 +748,13 @@ void SMTEncoder::endVisit(Identifier const& _identifier)
 		defineExpr(_identifier, m_context.state().thisAddress());
 		m_uninterpretedTerms.insert(&_identifier);
 	}
+	// Ignore the builtin abi, it is handled in FunctionCall.
+	// TODO: ignore MagicType in general (abi, block, msg, tx, type)
+	else if (auto magicType = dynamic_cast<MagicType const*>(_identifier.annotation().type); magicType->kind() == MagicType::Kind::ABI)
```

Comment:
We discussed to change this to `else if (!dynamic_cast<MagicType const*>(_identifier.annotation().type))`, but that warrants more changes and tests elsewhere, so deferring it.